libs/ngrx-toolkit/src/lib/flattening-operator.ts

```diff
@@ -0,0 +1,42 @@
+import {
+  concatMap,
+  exhaustMap,
+  mergeMap,
+  ObservableInput,
+  ObservedValueOf,
+  OperatorFunction,
+  switchMap,
+} from 'rxjs';
+
+export type RxJsFlatteningOperator = <T, O extends ObservableInput<unknown>>(
+  project: (value: T, index: number) => O,
+) => OperatorFunction<T, ObservedValueOf<O>>;
+
+/**
+ * A wrapper for an RxJS flattening operator.
+ * This wrapper informs about whether the operator has exhaust semantics or not.
+ */
+export type FlatteningOperator = {
```

Comment:
TODO: Works for now but we should think about something more generic than `exhaustSemantic` boolean for custom flatteners. The only thing we need to know, is if an operator skips an input completely. That means it doesn't even subscribe to it.

Reply:
Will be done in a separate issue